tests/test_deserialization.py

```diff
@@ -38,6 +38,24 @@ def test_tuple_field_deserialize_none_to_none(self):
         field = fields.Tuple([fields.String()], allow_none=True)
         assert field.deserialize(None) is None
 
+    def test_list_of_nested_allow_none_deserialize_none_to_none(self):
+        field = fields.List(fields.Nested(Schema(), allow_none=True))
+        assert field.deserialize([None, {}]) == [None, {}]
+
+    def test_nested_multiple_allow_none_deserialize_none_to_none(self):
+        field = fields.Nested(Schema(), allow_none=True, many=True)
+        assert field.deserialize([None, {}]) == [None, {}]
```

Comment:
The `allow_none` argument on `Nested` applies to the `Nested` container, not its children. Depending on `many` this is either a `List`, or a pass through container.

https://marshmallow.readthedocs.io/en/stable/api_reference.html#marshmallow.fields.Nested

Use `List(Nested(allow_none))` instead. #779 is an RFC that is amassing reasons `Nested(many)` should be deprecated. I will add this to the list.